types/got/index.d.ts

```diff
@@ -110,6 +110,8 @@ declare namespace got {
         followRedirect?: boolean;
         decompress?: boolean;
         useElectronNet?: boolean;
+        throwHttpErrors?: boolean;
+        cache?: object;
```

Comment:
I think that we should type this interface, from what I can gather, it should look something like:

```typescript
interface KeyvStorage {
  get (key: string): string | Promise<string>
  set (key: string, value: string): void | Promise<void>
  has (key: string): boolean | Promise<boolean>
  delete (key: string): void | Promise<void>
  clear (): void | Promise<void>
}
```

Reply:
IC, Will work on that! There are also this Keyv types defined [here](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/keyv/index.d.ts), I'll see if I can make them more consistent.

Reply:
I just approved another PR that typed this as `cache?: Map<string, any>;`, didn't really think about this at all since there was a lot of other changes as well 🤦‍♂️ 

I would still very much appreciate a PR that tidies this up with a proper interface, and also adds it to the Keyv typings.

Maybe the best would be if the Keyv typings could export the interface? and then we can depend and import it here?

Reply:
Sorry for the late reply! I got distracted for some errands. Please see the comment below for what I've done 😄